scripts/ci-test-summary.cjs

```diff
@@ -0,0 +1,130 @@
+        const testPath = test.titlePath();
+        const testName = testPath.pop();
+        let resultPath = this.testResult.failedTests;
+        testPath.forEach((item, index) => {
+          if (index === testPath.length - 1) {
+            if (!resultPath.hasOwnProperty(item)) {
+              resultPath[item] = [];
+            }
+            resultPath[item].push(testName);
+          }
+          if (!resultPath.hasOwnProperty(item)) {
+            resultPath[item] = {};
+          }
+          resultPath = resultPath[item];
+        });
+      })
+      .once(EVENT_RUN_END, () => {
+        this.testResult.stats = {
+          passed: stats.passes,
+          failed: stats.failures,
+          total: stats.passes + stats.failures,
+        };
+        this.writeSummary();
+      });
+  }
+
+  async writeSummary() {
+    try {
+      if (this.testResult.stats.failed > 0) {
+        this.summary = this.summary.addHeading('Failed tests', 3);
+        this.summary = this.summary
+          .addRaw(this.generateAccordian('', null, this.testResult.failedTests))
+          .addSeparator();
+      }
+      this.writeStatsTable();
+      this.writeProgressBar();
+      // Resettig GITHUB_STEP_SUMMARY, since unit tests might overwrite it
+      process.env.GITHUB_STEP_SUMMARY = this.destination;
+      await this.summary.write();
+    } catch (e) {
+      console.error(e);
+    }
+  }
+
+  generateAccordian(suitePrefix, suiteName, suiteObject) {
+    if (typeof suiteObject !== 'object') {
+      // Item is a test, not a suite
+      return `${suitePrefix}❌ ${suiteObject}<br />`;
+    }
+    let detailsContent = '';
+
+    // Generate accordian for each item in suite
+    Object.keys(suiteObject).forEach(
+      (item) =>
+        (detailsContent =
+          detailsContent +
+          this.generateAccordian(
+            `${suitePrefix}${this.suiteIndenter}`,
+            item,
+            suiteObject[item]
+          ))
+    );
+
+    // First level does not require accordian
+    if (suiteName === null) {
+      return detailsContent;
+    }
+
+    // Return an accordian for the suite
+    return `
+      <details>
+        <summary>${suitePrefix}${suiteName}</summary>
+        ${detailsContent}
+      </details>
+    `;
+  }
+
+  writeStatsTable() {
+    this.summary.addHeading('Stats', 3);
+    this.summary = this.summary.addTable([
+      ['Total number of tests', '' + this.testResult.stats.total],
+      ['Passed ✅', '' + this.testResult.stats.passed],
+      ['Failed ❌', '' + this.testResult.stats.failed],
+    ]);
+  }
+
+  writeProgressBar() {
+    let percentagePassed = Math.floor(
+      (this.testResult.stats.passed / this.testResult.stats.total) * 100
+    );
+    this.summary = this.summary.addRaw(`
+      <div style="width: 50%; background-color: red; text-align: center; color: white;">
+        <div style="width: ${percentagePassed}%; height: 18px; background-color: green;"></div>
+      </div>  
+    `);
```

Comment:
Thanks for your PR @Vedu1996! I haven't done an extensive review yet but was quite curious about the result so I started the actions for you. One thing that I noticed is that it's adding this section to the summary. Is there a specific reason you're adding this? Currently, it results in a simple code block with some HTML but nothing visual.